torch/csrc/dynamo/guards.cpp

```diff
@@ -836,6 +836,29 @@ class ID_MATCH : public LeafGuard {
   intptr_t _expected;
 };
 
+class EQUALS_MATCH : public LeafGuard {
+ public:
+  EQUALS_MATCH(py::object value, py::object verbose_code_parts)
+      : LeafGuard(verbose_code_parts),
+        _value(value),
+        _value_type(Py_TYPE(value.ptr())) {}
+
+  bool check_nopybind(PyObject* value) override { // borrowed ref
+    return Py_TYPE(value) == _value_type &&
+        PyObject_RichCompareBool(value, _value.ptr(), Py_EQ);
```

Comment:
Need to check for NULL return, add a fast path for ptr equals.

Reply:
Done. Ready for re-review.